src/Application/Messenger/Middleware/AddOrLogMessageId.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Messenger\Middleware;
+
+use MatchBot\Application\Messenger\DonationUpserted;
+use MatchBot\Application\Messenger\Stamp\MessageId;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+
+readonly class AddOrLogMessageId implements MiddlewareInterface
+{
+    public function __construct(private LoggerInterface $logger)
+    {
+    }
+
+    /** @psalm-suppress PossiblyUnusedReturnValue sig must match interface */
```

Comment:
Somehow the addition of the unit test turns this into an error, unless I suppress. Not clear why given the file is also new and so presumably not in the baseline. Removing the return type isn't possible as it violates the interface.

Reply:
I think for things like this I normally just put a comment to say that its called by the framework (or Symfony Messenger in this case), not by us.

Reply:
Right – made it a little clearer in d050e112dd784978c0565f37b49a575f9c8205f7. It's still odd to me that it is the presence of our test call that causes the error and that zero calls in our codebase is allowed – no `PossiblyUnusedMethod` in either case.